src/main/kotlin/org/jqassistant/tooling/intellij/plugin/data/report/ReportProviderService.kt

```diff
@@ -7,17 +7,45 @@ import com.intellij.openapi.diagnostic.thisLogger
 import com.intellij.openapi.project.BaseProjectDirectories.Companion.getBaseDirectories
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.VirtualFileManager
 import org.jqassistant.schema.report.v2.GroupType
 import org.jqassistant.schema.report.v2.JqassistantReport
 import org.jqassistant.schema.report.v2.ReferencableRuleType
+import org.jqassistant.tooling.intellij.plugin.common.JQAssistantPluginDisposable
 import java.io.File
 
+typealias ReportChangedListener = (rootDirectory: VirtualFile) -> Unit
+
```

Comment:
```suggestion
fun interface ReportChangedListener {
   fun invoke(rootDirectory: VirtualFile)
}
```

[SAM interfaces](https://kotlinlang.org/docs/fun-interfaces.html) seem like the better fit here.